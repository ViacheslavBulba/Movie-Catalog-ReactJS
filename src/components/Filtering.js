import React from 'react';
import '../css/Filtering.css';

export default function Filtering() {
    throw new Error('I crashed!');

    return (
        <div className="filter-container">
            <button className="filter-option">ALL</button>
            <button className="filter-option">DOCUMENTARY</button>
            <button className="filter-option">COMEDY</button>
            <button className="filter-option">HORROR</button>
            <button className="filter-option">CRIME</button>
        </div>
    );
}
