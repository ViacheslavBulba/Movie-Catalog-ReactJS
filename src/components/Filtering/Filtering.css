.filter-option {
    padding: 0;
    margin-right: 35px;
    height: 40px;
    font-family: 'Source Sans Pro';
    font-size: 18px;
    color: #ffffff;
    background-color: #232323;
    text-transform: uppercase;
}

.filter-active {
    border-bottom: 3px solid #fff;
}

.dropdown-label {
    margin-right: 20px;
    color: #555555;
}

@media screen and (max-width: 704px) {
    .filter-container {
        display: none
    }
}