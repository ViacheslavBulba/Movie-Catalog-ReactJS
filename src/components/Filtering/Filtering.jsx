import React from 'react';
import './Filtering.css';
import { useSelector } from 'react-redux';
import { setFilterByGenres, thunkedSetMovies } from '../../store/actions';
import store from '../../store/store';

export default function Filtering() {
  const genresFilter = useSelector((state) => state.filerByGenres);

  const onFilterChange = (value) => {
    let filters = genresFilter;
    if (value === 'All') {
      filters = [];
    } else if (filters.includes(value)) {
      filters = filters.filter((k) => k !== value);
    } else {
      filters = [...filters, value];
    }
    store.dispatch(setFilterByGenres(filters));
    store.dispatch(thunkedSetMovies());
  };

  return (
    <div className="filter-container">
      <button
        className={
                    `filter-option ${
                      genresFilter.length === 0 ? 'filter-active' : ''}`
                }
        onClick={() => onFilterChange('All')}
        type="button"
      >
        All
      </button>
      <button
        className={
                    `filter-option ${
                      genresFilter.includes('Drama') ? 'filter-active' : ''}`
                }
        onClick={() => onFilterChange('Drama')}
        type="button"
      >
        Drama
      </button>
      <button
        className={
                    `filter-option ${
                      genresFilter.includes('Fantasy') ? 'filter-active' : ''}`
                }
        onClick={() => onFilterChange('Fantasy')}
        type="button"
      >
        Fantasy
      </button>
      <button
        className={
                    `filter-option ${
                      genresFilter.includes('Comedy') ? 'filter-active' : ''}`
                }
        onClick={() => onFilterChange('Comedy')}
        type="button"
      >
        Comedy
      </button>
      <button
        className={
                    `filter-option ${
                      genresFilter.includes('Action') ? 'filter-active' : ''}`
                }
        onClick={() => onFilterChange('Action')}
        type="button"
      >
        Action
      </button>
    </div>
  );
}
