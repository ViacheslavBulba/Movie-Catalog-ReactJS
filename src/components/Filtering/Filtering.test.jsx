import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Filtering from './Filtering';

const mockStore = configureStore();

const initialState = {
  pending: false,
  movies: [],
  error: null,
  sortBy: 'release_date',
  sortOrder: 'desc',
  filerByGenres: [],
  search: '',
  searchBy: 'title',
  movieToOverview: null,
};

const store = mockStore(initialState);

test('snapshot test for filtering', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Filtering />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
