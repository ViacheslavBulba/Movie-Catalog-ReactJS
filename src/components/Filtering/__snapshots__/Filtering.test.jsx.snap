// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for filtering 1`] = `
<div
  className="filter-container"
>
  <button
    className="filter-option filter-active"
    onClick={[Function]}
    type="button"
  >
    All
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
    type="button"
  >
    Drama
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
    type="button"
  >
    Fantasy
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
    type="button"
  >
    Comedy
  </button>
  <button
    className="filter-option "
    onClick={[Function]}
    type="button"
  >
    Action
  </button>
</div>
`;
